### ISSUE TITLE
bind doesn't cross-compile (bind.dnsutils)

### ISSUE THREAD

**Issue by flokli** [NixOS/nixpkgs#90727]
NixOS adds `bind.dnsutils` to `environment.systemPackages`, but `bind` doesn't cross-compile:

```
unpacking sources
unpacking source archive /nix/store/z7k13yikcyn4v3llwabihjvypgmbfip3-bind-9.14.12.tar.gz
source root is bind-9.14.12
setting SOURCE_DATE_EPOCH to timestamp 1588764061 of file bind-9.14.12/srcid
patching sources
applying patch /nix/store/ac6p07ypkry57w6gs8h6a7b4m6rpccsr-dont-keep-configure-flags.patch
patching file bin/named/include/named/globals.h
patching file bin/named/main.c
Hunk #1 succeeded at 470 (offset 11 lines).
Hunk #2 succeeded at 1041 (offset 38 lines).
applying patch /nix/store/zrby82jw1g72gi9vmj954dg23glvl76x-remove-mkdir-var.patch
patching file Makefile.in
Hunk #1 succeeded at 50 (offset -3 lines).
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --prefix=/nix/store/abhxkl7ij7wjbmdcpalz0zbq12gynf21-bind-9.14.12-armv7l-unknown-linux-gnueabihf --bindir=/nix/store/abhxkl7ij7wjbmdcpalz0zbq12gynf21-bind-9.14.12-armv7l-unknown-linux-gnueabihf/bin --sbindir=/nix/store/abhxkl7ij7wjbmdcpalz0zbq12gynf21-bind-9.14.12-armv7l-unknown-linux-gnueabihf/sbin --includedir=/nix/store/cvlgyipkr2c56wkf81l6i8ahnkmqk8r1-bind-9.14.12-armv7l-unknown-linux-gnueabihf-dev/include --oldincludedir=/nix/store/cvlgyipkr2c56wkf81l6i8ahnkmqk8r1-bind-9.14.12-armv7l-unknown-linux-gnueabihf-dev/include --mandir=/nix/store/912ppispk1mfl822wlzysrq4i92ran6n-bind-9.14.12-armv7l-unknown-linux-gnueabihf-man/share/man --infodir=/nix/store/abhxkl7ij7wjbmdcpalz0zbq12gynf21-bind-9.14.12-armv7l-unknown-linux-gnueabihf/share/info --docdir=/nix/store/abhxkl7ij7wjbmdcpalz0zbq12gynf21-bind-9.14.12-armv7l-unknown-linux-gnueabihf/share/doc/bind --libdir=/nix/store/5rbmwk116fh1f9dn0ahi0l9hjjrlvl5b-bind-9.14.12-armv7l-unknown-linux-gnueabihf-lib/lib --libexecdir=/nix/store/5rbmwk116fh1f9dn0ahi0l9hjjrlvl5b-bind-9.14.12-armv7l-unknown-linux-gnueabihf-lib/libexec --localedir=/nix/store/5rbmwk116fh1f9dn0ahi0l9hjjrlvl5b-bind-9.14.12-armv7l-unknown-linux-gnueabihf-lib/share/locale --localstatedir=/var --with-libtool --with-libxml2=/nix/store/a66c8fskd5s6xkc2q43rsvj0g2vs0vy4-libxml2-2.9.10-armv7l-unknown-linux-gnueabihf-dev --with-openssl=/nix/store/9108i5x1qlyncz78hjyn8k0nkd8kzmxs-openssl-1.1.1g-armv7l-unknown-linux-gnueabihf-dev --without-python --without-atf --without-dlopen --without-docbook-xsl --without-gssapi --without-idn --without-idnlib --without-lmdb --without-libjson --without-pkcs11 --without-purify --with-randomdev=/dev/random --with-ecdsa --with-gost --without-eddsa --with-aes --with-libcap=/nix/store/za0gfm04zfvr2jl3z7xmxmz28irnp4zb-libcap-2.27-armv7l-unknown-linux-gnueabihf-dev --build=x86_64-unknown-linux-gnu --host=armv7l-unknown-linux-gnueabihf
configure: WARNING: unrecognized options: --without-atf, --without-idn, --without-idnlib, --with-randomdev, --with-ecdsa, --with-gost, --without-eddsa, --with-aes, --with-libcap
checking build system type... x86_64-unknown-linux-gnu
checking host system type... armv7l-unknown-linux-gnueabihf
checking for armv7l-unknown-linux-gnueabihf-gcc... armv7l-unknown-linux-gnueabihf-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether armv7l-unknown-linux-gnueabihf-gcc accepts -g... yes
checking for armv7l-unknown-linux-gnueabihf-gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... armv7l-unknown-linux-gnueabihf-gcc -E
checking for grep that handles long lines and -e... /nix/store/ixmnsf3qlnyxbpfz1p2nwpr420dmggx0-gnugrep-3.4/bin/grep
checking for egrep... /nix/store/ixmnsf3qlnyxbpfz1p2nwpr420dmggx0-gnugrep-3.4/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether make sets $(MAKE)... yes
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/b9g5nci9a8y4px65g4wq0gx3cvvwh9sm-gnused-4.8/bin/sed
checking for fgrep... /nix/store/ixmnsf3qlnyxbpfz1p2nwpr420dmggx0-gnugrep-3.4/bin/grep -F
checking for ld used by armv7l-unknown-linux-gnueabihf-gcc... armv7l-unknown-linux-gnueabihf-ld
checking if the linker (armv7l-unknown-linux-gnueabihf-ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... armv7l-unknown-linux-gnueabihf-nm
checking the name lister (armv7l-unknown-linux-gnueabihf-nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-unknown-linux-gnu file names to armv7l-unknown-linux-gnueabihf format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for armv7l-unknown-linux-gnueabihf-ld option to reload object files... -r
checking for armv7l-unknown-linux-gnueabihf-objdump... armv7l-unknown-linux-gnueabihf-objdump
checking how to recognize dependent libraries... pass_all
checking for armv7l-unknown-linux-gnueabihf-dlltool... no
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for armv7l-unknown-linux-gnueabihf-ar... armv7l-unknown-linux-gnueabihf-ar
checking for archiver @FILE support... @
checking for armv7l-unknown-linux-gnueabihf-strip... armv7l-unknown-linux-gnueabihf-strip
checking for armv7l-unknown-linux-gnueabihf-ranlib... armv7l-unknown-linux-gnueabihf-ranlib
checking for gawk... gawk
checking command to parse armv7l-unknown-linux-gnueabihf-nm output from armv7l-unknown-linux-gnueabihf-gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/4dbvbaw2wgfgr021m6y6ay9qajiajsc5-coreutils-8.31/bin/dd
checking how to truncate binary pipes... /nix/store/4dbvbaw2wgfgr021m6y6ay9qajiajsc5-coreutils-8.31/bin/dd bs=4096 count=1
checking for armv7l-unknown-linux-gnueabihf-mt... no
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if armv7l-unknown-linux-gnueabihf-gcc supports -fno-rtti -fno-exceptions... no
checking for armv7l-unknown-linux-gnueabihf-gcc option to produce PIC... -fPIC -DPIC
checking if armv7l-unknown-linux-gnueabihf-gcc PIC flag -fPIC -DPIC works... yes
checking if armv7l-unknown-linux-gnueabihf-gcc static flag -static works... no
checking if armv7l-unknown-linux-gnueabihf-gcc supports -c -o file.o... yes
checking if armv7l-unknown-linux-gnueabihf-gcc supports -c -o file.o... (cached) yes
checking whether the armv7l-unknown-linux-gnueabihf-gcc linker (armv7l-unknown-linux-gnueabihf-ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for a BSD-compatible install... /nix/store/4dbvbaw2wgfgr021m6y6ay9qajiajsc5-coreutils-8.31/bin/install -c
checking whether ln -s works... yes
checking for a POSIX-compatible shell... /nix/store/n313xks5ym0s0a5v8a5285rmnmvy6ms9-bash-4.4-p23/bin/bash
checking for armv7l-unknown-linux-gnueabihf-pkg-config... no
checking for pkg-config... no
checking whether to enable fuzzing mode... no
checking for ar... /nix/store/ivmds12g88jbjf8famgwkj36a4rrfcy6-binutils-2.31.1/bin/ar
checking for etags... no
checking for emacs-etags... no
checking for perl5... no
checking for perl... /nix/store/jp5bbs9xy2ag3gvbdqaxc5hka1i7r1m1-perl-5.30.2/bin/perl
checking for Python support... disabled
checking for armv7l-unknown-linux-gnueabihf-gcc... (cached) armv7l-unknown-linux-gnueabihf-gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether armv7l-unknown-linux-gnueabihf-gcc accepts -g... (cached) yes
checking for armv7l-unknown-linux-gnueabihf-gcc option to accept ISO C89... (cached) none needed
checking for armv7l-unknown-linux-gnueabihf-gcc option to accept ISO C99... none needed
checking for ANSI C header files... (cached) yes
checking for fcntl.h... yes
checking for regex.h... yes
checking for sys/time.h... yes
checking for unistd.h... (cached) yes
checking for sys/mman.h... yes
checking for sys/sockio.h... no
checking for sys/select.h... yes
checking for sys/param.h... yes
checking for sys/sysctl.h... yes
checking for net/if6.h... no
checking for sys/socket.h... yes
checking for net/route.h... yes
checking for linux/netlink.h... yes
checking for linux/rtnetlink.h... yes
checking threads.h usability... yes
checking threads.h presence... yes
checking for threads.h... yes
checking for C11 Thread-Local Storage using thread_local... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for working volatile... yes
checking for flexible array members... yes
checking for yield instruction support... yes
checking for sysctlbyname... no
checking for mmap... yes
checking for seteuid... yes
checking for setresuid... yes
checking for setegid... yes
checking for setresgid... yes
checking for size_t... yes
checking for ssize_t... yes
checking for uintptr_t... yes
checking whether time.h and sys/time.h may both be included... yes
checking for uname... yes
checking for GCC noreturn attribute... yes
checking for kqueue... no
checking for epoll_create1... yes
checking sys/devpoll.h usability... no
checking sys/devpoll.h presence... no
checking for sys/devpoll.h... no
checking devpoll.h usability... no
checking devpoll.h presence... no
checking for devpoll.h... no
checking whether byte ordering is bigendian... no
checking for arc4random... no
checking for arc4random_buf... no
checking for arc4random_uniform... no
checking for getrandom... yes
checking whether armv7l-unknown-linux-gnueabihf-gcc is Clang... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for sigwait... yes
checking for pthread_attr_getstacksize... yes
checking for pthread_attr_setstacksize... yes
checking for PTHREAD_MUTEX_ADAPTIVE_NP... using adaptive lock type
checking sched.h usability... yes
checking sched.h presence... yes
checking for sched.h... yes
checking for library containing sched_yield... none required
checking for sched_yield... yes
checking for pthread_yield... yes
checking for pthread_yield_np... no
checking sys/cpuset.h usability... no
checking sys/cpuset.h presence... no
checking for sys/cpuset.h... no
checking sys/procset.h usability... no
checking sys/procset.h presence... no
checking for sys/procset.h... no
checking for pthread_setaffinity_np... yes
checking for cpuset_setaffinity... no
checking for processor_bind... no
checking for sched_setaffinity... yes
checking for pthread_setname_np... yes
checking for pthread_set_name_np... no
checking for pthread_np.h... no
checking for flockfile... yes
checking for getc_unlocked... yes
checking for sysconf... yes
checking for libtool... yes
checking for openssl/ssl.h in /nix/store/9108i5x1qlyncz78hjyn8k0nkd8kzmxs-openssl-1.1.1g-armv7l-unknown-linux-gnueabihf-dev... yes
checking whether compiling and linking against OpenSSL works... yes
checking for OpenSSL >= 1.0.0 or LibreSSL... yes
checking for CRYPTO_zalloc... yes
checking for EVP_CIPHER_CTX_new... yes
checking for EVP_CIPHER_CTX_free... yes
checking for EVP_MD_CTX_new... yes
checking for EVP_MD_CTX_free... yes
checking for EVP_MD_CTX_reset... yes
checking for HMAC_CTX_new... yes
checking for HMAC_CTX_free... yes
checking for HMAC_CTX_reset... yes
checking for HMAC_CTX_get_md... yes
checking for ECDSA_sign... yes
checking for ECDSA_verify... yes
checking for ECDSA P-256 support... yes
checking for ECDSA P-384 support... yes
checking for Ed25519 support... yes
checking for Ed448 support... yes
checking for EVP_sha1... yes
checking for EVP_sha224... yes
checking for EVP_sha256... yes
checking for EVP_sha384... yes
checking for EVP_sha512... yes
checking for EVP_aes_128_ecb... yes
checking for EVP_aes_192_ecb... yes
checking for EVP_aes_256_ecb... yes
checking for DH_get0_key... yes
checking for ECDSA_SIG_get0... yes
checking for RSA_set0_key... yes
checking whether to enable FIPS mode in OpenSSL library... no
checking for PKCS11 for Public-Key Cryptography... no
checking for clock_gettime... yes
checking for GSSAPI library... disabled
checking for lmdb library... no
checking for libxml2 library... yes
checking for json library... no
checking for zlib library... no
checking for smf_enable_instance in -lscf... no
checking if "armv7l-unknown-linux-gnueabihf-gcc" supports -fno-strict-aliasing... yes
checking if "armv7l-unknown-linux-gnueabihf-gcc" supports -fno-delete-null-pointer-checks... yes
checking whether to use purify... no
checking whether to use gperftools profiler... no
configure: WARNING: Internal symbol table does not work with libtool.  Disabling symbol table.
checking for IPv6 structures... yes
checking for strlcpy... no
checking for strlcat... no
configure: checking for readline with -ledit
checking for readline... no
configure: checking for readline with -ledit -lterminfo
checking for readline... no
configure: checking for readline with -ledit -ltermcap
checking for readline... no
configure: checking for readline with -ledit -lncurses
checking for readline... no
configure: checking for readline with -ledit -lcurses
checking for readline... no
configure: checking for readline with -lreadline
checking for readline... (cached) no
configure: checking for readline with -lreadline -lterminfo
checking for readline... no
configure: checking for readline with -lreadline -ltermcap
checking for readline... no
configure: checking for readline with -lreadline -lncurses
checking for readline... no
configure: checking for readline with -lreadline -lcurses
checking for readline... no
checking for chroot... yes
checking whether to enable Linux capabilities... yes
checking sys/capability.h usability... yes
checking sys/capability.h presence... yes
checking for sys/capability.h... yes
checking for library containing cap_set_proc... -lcap
checking sys/un.h usability... yes
checking sys/un.h presence... yes
checking for sys/un.h... yes
checking for tzset... yes
checking for optarg declaration... yes
checking for st_mtim.tv_nsec... checking for if_nametoindex... yes
checking for nanosleep... yes
checking for usleep... yes
checking for explicit_bzero... yes
checking stdatomic.h usability... yes
checking stdatomic.h presence... yes
checking for stdatomic.h... yes
checking for memory model aware atomic operations... stdatomic.h
checking whether -latomic is needed for 64-bit stdatomic.h functions... no
checking uchar.h usability... yes
checking uchar.h presence... yes
checking for uchar.h... yes
checking compiler support for __builtin_unreachable()... yes
checking compiler support for __builtin_expect... yes
checking compiler support for __builtin_clz... yes
checking for latex... no
checking for pdflatex... no
checking for dblatex... no
checking for w3m... no
checking for pandoc... pandoc
checking for xsltproc... xsltproc
checking for xmllint... xmllint
checking for doxygen... doxygen
checking for curl... curl
checking for Docbook-XSL path... no
checking for html/docbook.xsl... "not found"
checking for xhtml/docbook.xsl... "not found"
checking for manpages/docbook.xsl... "not found"
checking for html/chunk.xsl... "not found"
checking for xhtml/chunk.xsl... "not found"
checking for html/chunktoc.xsl... "not found"
checking for xhtml/chunktoc.xsl... "not found"
checking for html/maketoc.xsl... "not found"
checking for xhtml/maketoc.xsl... "not found"
checking for cmocka >= 1.0.0... no
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking for setlocale... yes
using default tuning
checking whether to enable query trace logging... no
checking for librpz __attribute__s... yes
checking for library containing librpz_client_create... no
checking for Postgres DLZ driver... no
checking for MySQL DLZ driver... no
checking for Berkeley DB DLZ driver...... no
checking for file system DLZ driver... no
checking for LDAP DLZ driver... no
checking for ODBC DLZ driver... no
checking for stub DLZ driver... no
checking contributed DLZ drivers... no
configure: error: BUILD_CC not set
builder for '/nix/store/ci87bdd8qpvgwx1kiy0ybggcmp7pcfxr-bind-9.14.12-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 1
```

cc @NixOS/exotic-platform-maintainers

**Comment by flokli** [comment]
This might actually already be fixed by 6cdbb1fb1febfd97ab4490fba461c9eb6453fa33, will take a look.

**Comment by flokli** [comment]
Indeed, I was on a slightly older checkout (d6d07f262b171bb1e415e1a06c52288af056a98d), now fixed (but gpg broke, argh!).

Sorry for the noise.
